Gems/Archive/Code/Include/Archive/Clients/ArchiveInterfaceStructs.inl

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+namespace Archive
+{
+    // Implement byte storage multipliers
+    inline namespace literals
+    {
+        constexpr AZ::u64 operator"" _kib(AZ::u64 value)
+        {
+            return value * (1 << 10);
+        }
+
+        constexpr AZ::u64 operator"" _mib(AZ::u64 value)
+        {
+            return value * (1 << 20);
+        }
+
+        constexpr AZ::u64 operator"" _gib(AZ::u64 value)
+        {
+            return value * (1 << 30);
+        }
+    }
+
+    // ArchiveHeader::TocOffsetU64
+    inline ArchiveHeader::TocOffsetU64::TocOffsetU64() = default;
+
+    // Constrain the TocOffset to be 512 or greater
+    inline ArchiveHeader::TocOffsetU64::TocOffsetU64(AZ::u64 offset)
+        : m_value(AZStd::max(offset, ArchiveDefaultBlockAlignment))
+    {
+    }
+
+    inline auto ArchiveHeader::TocOffsetU64::operator=(AZ::u64 offset) -> TocOffsetU64&
+    {
+        m_value = AZStd::max(offset, ArchiveDefaultBlockAlignment);
+        return *this;
+    }
```

Comment:
Because TocOffsetU64 is an inner class.
Therefore the suggested signature above wouldn't work as is `inline TocOffsetU64& ArchiveHeader::...`, as it would have to scope the TocOffsetU64 name a second time with `ArchiveHeader` class.

It would have to be
```suggestion
    inline ArchiveHeader::TocOffsetU64 ArchiveHeader::TocOffsetU64::operator=(AZ::u64 offset)
    {
        m_value = AZStd::max(offset, ArchiveDefaultBlockAlignment);
        return *this;
    }
```

Reply:
Ah, ok, thanks for the explanation! Just wasn't seeing the benefit of the other syntax, but I get it now.